### PR TITLE
Brian/#223 dev proxy

### LINKED ISSUE #223: Proxy requests in Dev to avoid CORS
Since Webpack is making requests from a different domain (localhost:7900) to the daemon, CORS was enabled.

Webpack Dev Server can proxy its requests through the daemon to avoid this issue.

### PR DESCRIPTION
**Pull request by brian-nguyen**
:tickets: **Ticket(s)**: Closes #223 

---

## :construction_worker: Changes

CORS only affects development due to Webpack Dev Server. Set up proxy config in package.json and webpack [config](https://webpack.js.org/configuration/dev-server/#devserver-proxy), and then removed CORS handling. As a result, all requests to the daemon are now made relative to the server root.

## :flashlight: Testing Instructions

Open the Webpack hosted page. Note that for testing, I disabled the cookie and admin checks on the daemon to isolate whether the `/validate` route was correctly proxied or not. Some more help is needed!

### CODE REVIEW

**Comment by brian-nguyen** [comment]
Removal of `this.url` from the get and post methods in [client.js](https://github.com/ubclaunchpad/inertia/blob/c78680f4c6a21842ceeb77e3af1e454357cebd12/daemon/web/client.js#L76) has resulted in eslint complaining. How should this be handled? Is the rule `class-methods-use-this` something that is strictly enforced?

File: daemon/inertiad/auth/permissions.go [hunk 1]
```diff
@@ -88,19 +87,6 @@ func (h *PermissionsHandler) Close() error {
 
 // nolint: gocyclo
 func (h *PermissionsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	// CORS for development
-	origin, err := url.Parse(r.Header.Get("Origin"))
```
Comment (bobheadxi) [inline daemon/inertiad/auth/permissions.go]: nice nice this was ugly 😅

File: daemon/web/webpack.config.js [hunk 2]
```diff
@@ -13,6 +13,9 @@ const config = {
     inline: true,
     contentBase: './public',
     publicPath: '/web/',
+    proxy: {
+      '/': { target: 'https://127.0.0.1:4303', secure: false },
```
Comment (bobheadxi) [inline daemon/web/webpack.config.js]: TIL 🔥

**Comment by PiggySpeed** [comment]
@brian-nguyen  @bobheadxi 

> Removal of this.url from the get and post methods in client.js has resulted in eslint complaining. How should this be handled? Is the rule class-methods-use-this something that is strictly enforced?

This warning is suggesting that we may want to convert the `get` method into a static because it does not depend on any information from the instance. In fact, by removing `this.url` from the `get` and `post` methods, we are not making use of any instance variables and can convert the whole thing to a bunch of module-level helper functions.

So here are some options:
1. remove the constructor, and convert all the methods to static ones; remove all uses of `this` in all other methods
2. extract all methods out into module-level helper functions (i.e. remove the class altogether)
I am okay with either one.

When it comes to eslint rules, I play on the conservative side and try to avoid changing them as much as possible.

**Comment by PiggySpeed** [comment]
> I think the web app is due for an overhaul anyway

Yes, we don't have to be very picky about the elegance of the codebase since the existing web app was just a quick prototype until we fleshed out more details about the interface.

File: daemon/web/client.js [hunk 3]
```diff
@@ -66,14 +66,15 @@ export default class InertiaClient {
    * @param {Object} params
    * @param {{[key]: string}} queryParams
    */
+  // eslint-disable-next-line
```
Comment (PiggySpeed) [inline daemon/web/client.js]: See previous comment regarding `this.url`.

**Comment by brian-nguyen** [comment]
@PiggySpeed @bobheadxi 

Sounds good. I just went ahead and made the methods all static and changed all usage to those static methods. Let me know if I missed anything

**Comment by bobheadxi** [comment]
paging @zhulia96 and @rogermyang who will be working on the web app a bit more on this dicussion

**Pull request merged**